### PR TITLE
[13.0][MIG] web m2x options

### PR DESCRIPTION
**Pull request by andres-pcg**
This includes changes requested by @Yajo to https://github.com/OCA/web/pull/1392
cc: @daramousk

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please fix Travis and runbot

**Comment by pedrobaeza** [comment]
TT19827

**Comment by yajo** [comment]
Please attend Travis' :x:. You have to follow the migration steps as explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0

File: web_m2x_options/static/src/js/form.js [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+        FieldMany2ManyTags = relational_fields.FieldMany2ManyTags,
+        FieldMany2One = relational_fields.FieldMany2One,
+        FormFieldMany2ManyTags = relational_fields.FormFieldMany2ManyTags;
+
+    var web_m2x_options = rpc.query({
+        model: "ir.config_parameter",
+        method: 'get_web_m2x_options',
+    });
+
+    var M2ODialog = Dialog.extend({
```
Comment (i-vyshnevska) [inline web_m2x_options/static/src/js/form.js:24]: M2ODialog now completely copy pasted to odoo I think you can remove it completly
https://github.com/odoo/odoo/blob/08540d7853d497e326182e822b359cd6fcb05d59/addons/web/static/src/js/fields/relational_fields.js#L38

Comment (andres-pcg) [reply web_m2x_options/static/src/js/form.js:24]: Great; thanks. but It can't be removed since the method is not being returned in the core.

**Comment by andres-pcg** [comment]
@Yajo @pedrobaeza this should be good to go. 👍 

Thanks a lot for your suggestion @Yajo.

**Review by AdriaGForgeFlow — CHANGES_REQUESTED** [review]
This module is missing a lot of files, also commit history is not preserved.

I don't know why travis is green...

**Comment by andres-pcg** [comment]
Thanks, @AdriaGForgeFlow . Missing files has been added.

**Review by AdriaGForgeFlow — CHANGES_REQUESTED** [review]
Adding this ir.config_parameter:
image

Gives the following error:
image

**Comment by andres-pcg** [comment]
@AdriaGForgeFlow Please double check since i just review the workflow again and it worked seamlessly.

Runbot instance:
http://3411602-1530-2f4c32.runbot2-3.odoo-community.org/web?debug=1#id=16&action=11&model=ir.config_parameter&view_type=form&cids=1&menu_id=4

Capture:
image

**Comment by AdriaGForgeFlow** [comment]
> @AdriaGForgeFlow Please double check since i just review the workflow again and it worked seamlessly.


Sorry I did not explain myself. The error does not appear when creating the ir.config_parameter but when trying to modify a Many2one field. See below:
Peek 2020-03-06 08-32

**Comment by andres-pcg** [comment]
> > @AdriaGForgeFlow Please double check since i just review the workflow again and it worked seamlessly.
> 
> Sorry I did not explain myself. The error does not appear when creating the ir.config_parameter but when trying to modify a Many2one field. 


@AdriaGForgeFlow Please describe steps to follow since i have just tested it again and i'm unable to reproduce it on runbot: 
http://3411932-12-0-de873e.runbot1-2.odoo-community.org/web?debug=true#id=&action=77&model=res.config.settings&view_type=form&menu_id=4

BXbtDuJR6E

**Comment by yajo** [comment]
Try to set it to `0` instead.

**Comment by AdriaGForgeFlow** [comment]
> @AdriaGForgeFlow Please describe steps to follow since i have just tested it again and i'm unable to reproduce it on runbot:
> http://3411932-12-0-de873e.runbot1-2.odoo-community.org/web?debug=true#id=&action=77&model=res.config.settings&view_type=form&menu_id=4

What you are showing is an Odoo v12. In v13 is not working.
Check here http://3412581-1530-2f4c32.runbot2-2.odoo-community.org/web?debug=1#id=&action=83&model=res.config.settings&view_type=form&cids=1&menu_id=4

**Comment by bodedra** [comment]
@andrp92 I have tested and I faced same error mentioned by @AdriaGForgeFlow 

I have tried three cases:

1. web_m2x_options.create = False
2. web_m2x_options.create = 0
3. Remove above both options.

Here is ref of 3rd test case.

out-14

It seems this break m2o drop down.

File: web_m2x_options/static/src/js/form.js [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+
+        _search: function(search_val) {
+            var def = new Promise(resolve => {
+                // Add options limit used to change number of selections record
+                // returned.
+                if (!_.isUndefined(self.ir_options["web_m2x_options.limit"])) {
+                    this.limit = parseInt(self.ir_options["web_m2x_options.limit"], 10);
+                }
+
+                if (typeof self.nodeOptions.limit === "number") {
```
Comment (TDu) [inline web_m2x_options/static/src/js/form.js:146]:
```suggestion
                if (typeof this.nodeOptions.limit === "number") {
```

File: web_m2x_options/static/src/js/form.js [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+                    self.limit = self.nodeOptions.limit;
```
Comment (TDu) [inline web_m2x_options/static/src/js/form.js:147]:
```suggestion
                    this.limit = this.nodeOptions.limit;
```

File: web_m2x_options/static/src/js/form.js (hunk 2 above)
Comment (TDu) [inline web_m2x_options/static/src/js/form.js:146]:
```suggestion
                if (typeof this.nodeOptions.limit === "number") {
```

**Review by TDu — CHANGES_REQUESTED** [review]
In the `_search` function self is the window object ! See my two suggested changes, but this is recurring further down the function.

**Comment by bodedra** [comment]
@TDu Thanks for your suggestion. I have tried in my local environment but it doesn't help. I face same error.

**Comment by bodedra** [comment]
Inside console it throws `TypeError: self.ir_options is undefined`

I am trying to resolve it.

**Comment by bodedra** [comment]
JFI, I have find the root cause and will debug more. If all case work, I will submit PR.

**Comment by i-vyshnevska** [comment]
I had similar issues, you can check my solution for this https://github.com/OCA/web/pull/1532/commits/628efb0147e63fcd5fd3742010a16ab1c08a4dee

**Comment by bodedra** [comment]
PR Ref: https://github.com/OCA/web/pull/1564

**Comment by bodedra** [comment]
@i-vyshnevska Thanks for sharing your valuable situation. I have tried with small changes and verified output. It works like a charm.

**Comment by dreispt** [comment]
Superseded by #1564

**Pull request closed without merging**